File: Array/zhanchengguo/0628-平均值子数组/Main.java
package com.test.ktapplication;


import android.util.Log;

public class Main {

    void main() {

        int[] a = {2, 2, 2, 2, 5, 5, 5, 8};
        int result = numOfSubarrays(a, 3, 4);
        Log.e("result is", result + "");
    }

    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int length = arr.length;
        int sum = 0;
        int count = 0;
        for (int i = 0; i < k; i++) {
            sum += arr[i];
        }
        if (sum >= k * threshold) {
            count++;
        }
        if (k == length) {
            return count;
        }
        for (int j = 0; j < length - k; j++) {
            sum -= arr[j];
            sum += arr[j + k];
            if (sum >= k * threshold) {
                count++;
            }
        }
        return count;
    }

}

File: Array/zhanchengguo/0628-平均值子数组/main.kt
package com.test.ktapplication


fun main() {
    val s = intArrayOf(2, 2, 2, 2, 5, 5, 5, 8)
    var result = numOfSubarrays(s, 3, 4)
    print("result is ${result}")
}

fun numOfSubarrays(arr: IntArray, k: Int, threshold: Int): Int {
    var sum = 0
    var count = 0
    val length = arr.size
    for (i in 0 until k) {
        sum += arr[i]
    }
    if (sum >= k * threshold) {
        count++
    }
    if (length == k) {
        return count
    }
    for (j in 0 until length - k) {
        sum -= arr[j]
        sum += arr[j + k]
        if (sum >= k * threshold) {
            count++
        }
    }
    return count
}
